src/SFA.DAS.EAS.Portal/Application/Commands/AddReservationCommand.cs

```diff
@@ -0,0 +1,75 @@
+﻿using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using SFA.DAS.CosmosDb;
+using SFA.DAS.EAS.Portal.Database;
+using SFA.DAS.EAS.Portal.Database.Models;
+using SFA.DAS.Reservations.Messages;
+
+namespace SFA.DAS.EAS.Portal.Application.Commands
+{
+    public class AddReservationCommand
+    {
+        private readonly IAccountsRepository _accountsRepository;
+        private readonly ILogger<AddReservationCommand> _logger;
+
+        public AddReservationCommand(IAccountsRepository accountsRepository, ILogger<AddReservationCommand> logger)
+        {
+            _accountsRepository = accountsRepository;
+            _logger = logger;
+        }
+
+        public async Task Execute(ReservationCreatedEvent reservedFunding, string messageId, CancellationToken cancellationToken = default)
+        {
+            _logger.LogInformation("Executing AddReservationCommand");
```

Comment:
Do we need this log statement in here?  If it's a requirement then better to decorate the handler and put the logging in there. Keeps the purpose of the code clearer and less cluttered with noise..  Also makes unit testing simpler.

Reply:
We don't explicitly need any particular log statement and there is no requirement to log. However logging helps us see if the system is working correctly in the environments (which is very helpful when initially provisioning the environment and ironing out any teething issues), and it also helps us understand the causes of bugs.

I personally don't see an issue with adding log statements where they are most useful.

If we decide on verifying the log output in unit tests, then we'd have to verify the log output of whatever class the log line ends up in, so the complexity of the unit tests overall is constant. I tend to only unit test log output when there _is_ some sort of explicit requirement, such as auditing.